g2p_programs/models/managers/payment_manager.py

```diff
@@ -81,24 +83,37 @@ def prepare_payments(self, cycle):
         )
         # _logger.info("DEBUG! payments_to_create: %s", payments_to_create)
 
-        # Todo: Create payment batch
-        # batch = self.env["g2p.paymentbatch"].create()
-
         ctr = 0
+        vals = []
+        payments_to_add_ids = []
         for entitlement_id in entitlements_with_payments_to_create:
-            self.env["g2p.payment"].create(
+            payment = self.env["g2p.payment"].create(
                 {
                     "entitlement_id": entitlement_id.id,
                     "cycle_id": entitlement_id.cycle_id.id,
                     "amount_issued": entitlement_id.initial_amount,
                     "payment_fee": entitlement_id.transfer_fee,
                     "state": "issued",
                     # "account_number": self._get_account_number(entitlement_id),
-                    # "batch_id": batch.id,
                 }
             )
+            vals.append((4, payment.id))
+            payments_to_add_ids.append(payment.id)
             ctr += 1
```

Comment:
Is the construct `ctr = 0`/`ctr += 1`/`if ctr > 0` used for perfomance reasons instead of checking `if len(payments_to_add_ids) > 0` ?

Reply:
Done